spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/convert/MappingMongoConverterTests.java

```diff
@@ -478,4 +481,64 @@ public String toString() {
 					+ ", localTime=" + this.getLocalTime() + ", localDateTime=" + this.getLocalDateTime() + ")";
 		}
 	}
+
+	@Nested // GH-5065
+	@DisplayName("GH-5065: Empty Map Tests")
+    class EmptyMapTests {
+
+		@Test // GH-5065
+		@DisplayName("Passing test to indicate that the problem is not a more generic issue with maps in general.")
+		void readsEmptyMapCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
+			EmptyMapDocument target = converter.read(EmptyMapDocument.class, document);
+			assertThat(target.map).isNotNull().isEmpty();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an empty object as an empty map when using @DocumentReference.")
+		void readsEmptyMapWithDocumentReferenceCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
+			DocumentReferenceEmptyMapDocument target = converter.read(DocumentReferenceEmptyMapDocument.class, document);
+			assertThat(target.map).isNotNull().isEmpty();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an empty object as an empty map with a valis values property when using @DocumentReference(lazy = true).")
+		void readsEmptyMapWithLazyLoadedDocumentReferenceCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
```

Comment:
I think it would also be interesting to have test for `{ 'map' : null }` expecting `null` for the `target.map`.